conda/core/package_cache_data.py

```diff
@@ -644,83 +713,109 @@ def execute(self):
             # TODO: use logger
```

Comment:
It would be really nice to have a doc string in this method explaining what is going on at a high level.